jetty-servlet/src/main/java/org/eclipse/jetty/servlet/ServletContextHandler.java

```diff
@@ -1493,4 +1501,43 @@ public void destroy(Object o)
             decorator.destroy(o);
         }
     }
+
+    /**
+     * A utility class to hold a {@link ServletContainerInitializer} and implement the
+     * {@link ServletContainerInitializerCaller} interface so that the SCI is correctly
+     * started if an instance of this class is added as a bean to a {@link ServletContextHandler}.
+     */
+    public static class Initializer extends AbstractLifeCycle implements ServletContainerInitializerCaller
```

Comment:
I don't see any refactoring of the existing ContainerInitializer and ServletContainerInitializersStarter classes used by webapps - I would think that we could extend this ServletContextHandler.Initializer class, or have the other classes extend it. I don't like having 2 completely separate mechanisms.

Reply:
We have a single mechanism based on a single interface, `ServletContainerInitializerCaller`, that has two key implementations:
 1. The webapp impl that has a single caller that invokes all the SCIs it has discovered in the order that it wants.
 2. The Initializer impl that is just a convenience class to create your own custom caller.

To make the webapp impl use the Initializer, then it would have to make all discovered SCIs individually wrapped as Initializers and get into the business of sorting beans on the context....  That is doable, but I'm not sure it is worth it for the bugs that it will introduce.  I really don't want to mess with webapps SCI handling if it is not broken.

Reply:
But I thought that the ordering on the beans is that they are called in the order they are added? If so, then we can just have AnnotationConfiguration add them in the order that is established by applying any ordering: currently it just adds the ContainerInitializers to a list in the order determined by the ordering, so I don't see the difference between adding it to that list, and adding it to the  list of beans.

Reply:
Yes it is totally doable... but I just don't want to mess with something that isn't broken.... maybe 10.1?